lib/rack/cache/meta_store.rb

```diff
@@ -64,8 +64,7 @@ def store(request, response, entity_store)
         end
         response.headers['X-Content-Digest'] = digest
         response.headers['Content-Length'] = size.to_s unless response.headers['Transfer-Encoding']
-        # If the entitystore backend is a Noop, do not try to read the body from the backend, it always returns an empty array
-        response.body = entity_store.open(digest) || response.body unless entity_store.is_a? Rack::Cache::EntityStore::Noop
+        response.body = entity_store.open(digest) || response.body
```

Comment:
If you remove this conditional, response bodies will **never** (when using the ```Noop``` backend) be returned to client code, not even for responses that do not come from the cache!

IMHO the intended behavior of this backend is: 

- when the client performs a request, if the response is **not in the cache**, 
  - save the response metadata in the cache
  - return the response to the client, **with the body intact**. The client can check that the response does not come from the cache looking at X-Rack-Cache
- if the response **is cached**, return the cached headers and an empty body ```[]```. The client can check that the response comes from the cache looking at X-Rack-Cache and ignore its body.

If you remove the ```unless entity_store.is_a? Rack::Cache::EntityStore::Noop``` conditional, the response body will **always** be empty, even for responses that do not come from the cache. This would make the backend useless, in my opinion.